pyHalo/PresetModels/external.py

```diff
@@ -52,107 +70,106 @@ def DMFromGalacticus(z_lens,z_source,galacticus_hdf5,tree_index, kwargs_cdm,mass
 
     params_to_read = galacticus_params_additional + PARAMS_TO_READ_DEF
     
-    # we create a realization of only line-of-sight halos by setting sigma_sub = 0.0
-    # only include these halos if requested
-    kwargs_cdm['sigma_sub'] = 0.0
+    if isinstance(galacticus_hdf5,str):
+        nodedata = gutil.read_nodedata_galacticus(galacticus_hdf5,params_to_read=params_to_read)
+    elif isinstance(galacticus_hdf5, h5py.Group):
+        nodedata = gutil.hdf5_read_galacticus_nodedata(galacticus_hdf5,params_to_read=params_to_read)
+    else:
+        nodedata = galacticus_hdf5
+
+    z_lens = np.mean(nodedata[gutil.PARAM_Z_LAST_ISOLATED][np.logical_not(nodedata_filter_subhalos(nodedata,gutil))])
 
-    los_norm = 1 if include_field_halos else 0
-    los_norm = kwargs_cdm.get("LOS_normalization") if not kwargs_cdm.get("LOS_normalization") is None else los_norm
+    if tree_index is None:
+        tree_index = np.random.choice(np.unique(nodedata[gutil.PARAM_TREE_INDEX]))
 
+    # we create a realization of only line-of-sight halos by setting sigma_sub = 0.0
+    kwargs_cdm['sigma_sub'] = 0.0
+    kwargs_cdm["cone_opening_angle_arcsec"] = cone_opening_angle_arcsec
 
-    cdm_halos_LOS = CDM(z_lens, z_source, **kwargs_cdm,LOS_normalization=los_norm)
+    cdm_halos_LOS = CDM(z_lens, z_source, **kwargs_cdm)
     
     # get lens_cosmo class from class containing LOS objects; note that this will work even if there are no LOS halos
     lens_cosmo = cdm_halos_LOS.lens_cosmo
 
-    if isinstance(galacticus_hdf5,str):
-        nodedata = gutil.read_nodedata_galacticus(galacticus_hdf5,params_to_read=params_to_read)
-    else:
-        nodedata = gutil.hdf5_read_galacticus_nodedata(galacticus_hdf5,params_to_read=params_to_read)
-
-
-    #Set up for rotation of coordinates
-    #Specify the normal vector for the plane we are projecting onto, if user specified ensure the vector is normalized
-    nh = np.asarray((0,0,1)) if proj_plane_normal is None else proj_plane_normal / np.linalg.norm(proj_plane_normal)
-    nh_x,nh_y,nh_z = nh
+    theta,phi = np.random.random(2) * np.asarray((np.pi, 2 * np.pi))
 
-    theta = np.arccos(nh_z)
-    phi = np.sign(nh_y) * np.arccos(nh_x/np.sqrt(nh_x**2 + nh_y**2)) if nh_x != 0 or nh_y != 0 else 0
+    if not proj_plane_normal is None:
```

Comment:
I would change how proj_plane_normal is specified in the function call so that the orientation of the subhalo population can be randomly sampled. The sampling done in quadmodel is for individual parameters, so you'll need to break up proj_plane_normal into its components. 

I suggest adding three arguments to the function with default values projection_x=1, projection_y=0, projection_z=0. You then have the option of sampling values of projection_x, projection_y, and projection_z from a prior to generate random orientations.

Reply:
Done. I also changed the function to only accept angle arguments instead of angle arguments and vector arguments. This way the inputs are simpler, and passing in an angle is probably more natural when sampling from a distribution.